chirp/drivers/retevis_rb28.py

```diff
@@ -0,0 +1,717 @@
+
+#seekto 0x00AD;
+struct {
+u8 chnl_8:1,      // Busy Channel Lockout  8     00AD
+   chnl_7:1,      // Busy Channel Lockout  7
+   chnl_6:1,      // Busy Channel Lockout  6
+   chnl_5:1,      // Busy Channel Lockout  5
+   chnl_4:1,      // Busy Channel Lockout  4
+   chnl_3:1,      // Busy Channel Lockout  3
+   chnl_2:1,      // Busy Channel Lockout  2
+   chnl_1:1;      // Busy Channel Lockout  1
+u8 chnl_16:1,     // Busy Channel Lockout 16     00AE
+   chnl_15:1,     // Busy Channel Lockout 15
+   chnl_14:1,     // Busy Channel Lockout 14
+   chnl_13:1,     // Busy Channel Lockout 13
+   chnl_12:1,     // Busy Channel Lockout 12
+   chnl_11:1,     // Busy Channel Lockout 11
+   chnl_10:1,     // Busy Channel Lockout 10
+   chnl_9:1;      // Busy Channel Lockout 09
+u8 unused_af_7:1, //                             00AF
+   unused_af_6:1, //
+   chnl_22:1,     // Busy Channel Lockout 22
+   chnl_21:1,     // Busy Channel Lockout 21
+   chnl_20:1,     // Busy Channel Lockout 20
+   chnl_19:1,     // Busy Channel Lockout 19
+   chnl_18:1,     // Busy Channel Lockout 18
+   chnl_17:1;     // Busy Channel Lockout 17
+} bclo;
+"""
+
+CMD_ACK = b"\x06"
+
+BACKLIGHT_LIST = ["Off", "ON", "5S", "10S", "15S", "30S"]
+PFKEY_LIST = ["None", "Weather", "Warn", "Call", "Monitor"]
+SAVE_LIST = ["Normal", "Super"]
+TIMEOUTTIMER_LIST = ["%s seconds" % x for x in range(15, 195, 15)]
+VOICE_LIST = ["Off", "English"]
+VOXD_LIST = ["0.5", "1.0", "1.5", "2.0", "2.5", "3.0"]
+VOXL_LIST = ["OFF"] + ["%s" % x for x in range(1, 10)]
+
+SETTING_LISTS = {
+    "backlight": BACKLIGHT_LIST,
+    "pfkey_gt": PFKEY_LIST,
+    "pfkey_lt": PFKEY_LIST,
+    "save": SAVE_LIST,
+    "tot": TIMEOUTTIMER_LIST,
+    "voice": VOICE_LIST,
+    "voxd": VOXD_LIST,
+    "voxl": VOXL_LIST,
+    }
+
+PMR_TONES = list(chirp_common.TONES)
+[PMR_TONES.remove(x) for x in [ 69.3, 159.8, 165.5, 171.3, 177.3, 183.5,
+                               189.9, 196.6, 199.5, 206.5, 229.1, 254.1]]
+
+PMR_DTCS_CODES = list(chirp_common.DTCS_CODES)
+[PMR_DTCS_CODES.remove(x) for x in [ 36,  53, 122, 145, 212, 225, 246,
+                                    252, 255, 266, 274, 325, 332, 356,
+                                    446, 452, 454, 455, 462, 523, 526]]
+
+def _enter_programming_mode(radio):
+    serial = radio.pipe
+
+    _magic = radio._magic
+
+    try:
+        serial.write(_magic)
+        for i in range(1, 5):
+            ack = serial.read(1)
+            if ack == CMD_ACK:
+                break
+    except:
+        raise errors.RadioError("Error communicating with radio")
+
+    if not ack:
+        raise errors.RadioError("No response from radio")
+    elif ack != CMD_ACK:
+        raise errors.RadioError("Radio refused to enter programming mode")
+
+    try:
+        serial.write(b"\x02")
+        ident = serial.read(8)
+    except:
+        raise errors.RadioError("Error communicating with radio")
+
+    # check if ident is OK
+    itis = False
+    for fp in radio._fingerprint:
```

Comment:
Why not just do:
```
itis = ident in radio._fingerprint
```
? Is `ident` longer than all the fingerprints? Doesn't matter much of course. Another trick to do what you're doing but shorter is to use `else` on the for, which is only called if no break happens:
```
for fp in radio._fingerprint:
    if fp in ident:
        break
else:
    # No break ever got called, which means we didn't find it
    raise errors.RadioError('Radio identification failed')

# If we made it this far, we're good.
```

Reply:
It is what Pavel developed when he worked with me on the original btech.py driver. I like this, though. I will keep it in mind. Thanks for the "trick".